backend/app/services/intent_gate.py

```
@@ -0,0 +1,107 @@
+
+    def classify(self, query: str) -> IntentGateOut:
+        base = self._classify_rules(query)
+
+        # Deterministic-first: LLM is optional enhancement.
+        try:
+            if self.llm_service.is_enabled():
+                llm_out = self.llm_service.classify(query)
+                return self._merge_base_with_llm(base, llm_out, query)
+        except Exception:
+            # Fall back to deterministic rules for demo reliability.
+            pass
```

Ruff (0.15.12)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)